homeassistant/components/axis.py

```diff
@@ -287,27 +296,19 @@ def _write_config(hass, config):
         outfile.write(data)
 
 
-def event_initialized(event):
-    """Register event initialized on metadatastream here."""
-    hass = event.device_config('hass')
-    discovery.load_platform(hass,
-                            convert(event.topic, 'topic', 'platform'),
-                            DOMAIN, {'axis_event': event})
-
-
 class AxisDeviceEvent(Entity):
     """Representation of a Axis device event."""
 
-    def __init__(self, axis_event):
+    def __init__(self, event_config):
         """Initialize the event."""
-        self.axis_event = axis_event
-        self._event_class = convert(self.axis_event.topic, 'topic', 'class')
-        self._name = '{}_{}_{}'.format(self.axis_event.device_name,
-                                       convert(self.axis_event.topic,
-                                               'topic', 'type'),
+        self.axis_event = event_config[CONF_EVENT]
+        self._name = '{}_{}_{}'.format(event_config[CONF_NAME],
+                                       self.axis_event.event_type,
                                        self.axis_event.id)
+        self.location = event_config[ATTR_LOCATION]
         self.axis_event.callback = self._update_callback
 
+    @callback
```

Comment:
You can't call `schedule_update_ha_state` from an async context.

Reply:
Ok, will look in to this.

I'm not fully getting the async context thing yet.

Reply:
As I understand it, the chain of methods are not on the event loop so I will remove the @callback